operator-common/src/main/java/io/strimzi/operator/common/ReconnectingWatcher.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import io.strimzi.operator.common.operator.resource.AbstractWatchableNamespacedResourceOperator;
+
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+/**
+ * The Fabric8 Watcher which automatically reconnects when it is closed with an error.
+ *
+ * @param <T> The resource type
+ */
+public class ReconnectingWatcher<T extends HasMetadata> implements Watcher<T> {
```

Comment:
I think I would keep the name as it is. It would be able to watch anything. It's just that right now only the CrdOperator implements `AbstractWatchableNamespacedResourceOperator`. But that is just about our abstraction.

Reply:
Fair enough.